src/clients/AcrossConfigStoreClient/AcrossConfigStoreClient.ts

```diff
@@ -417,6 +432,37 @@ export class AcrossConfigStoreClient extends BaseAbstractClient {
         if (!isNaN(args.value)) {
           this.cumulativeMaxRefundCountUpdates.push(args);
         }
+      } else if (args.key === utf8ToHex(GLOBAL_CONFIG_STORE_KEYS.CHAIN_ID_INDICES)) {
+        // First remove out all the spaces
+        const rawChainIndices = String(args.value).replace(/\s/g, "");
+        // Sanity check to verify that this is a string representation of an array of
+        // positive integers. Let's confirm this via a regex.
+        if (!rawChainIndices.match(/^\[[0-9,]+\]$/)) {
+          this.logger.warn({ at: "ConfigStore", message: `The array ${rawChainIndices} is invalid.` });
+          // If not a valid array, skip.
+          continue;
+        }
+        // Parse this via JSON.parse. Since we've passed the regex check, we can
+        // be sure that this is a valid array of positive integers.
+        const chainIndices = JSON.parse(rawChainIndices) as number[];
+
+        // We now need to check that we're only appending positive integers to the
+        // chainIndices array on each update. If this isn't the case, we're going to
+        // need to skip this update & warn.
+        // Resolve the previous update. If there is no previous update, then we can
+        // assume that the default chain indices are being used. These default chain
+        // indices are [1, 10, 137, 288, 42161] (outlined in UMIP-157)
+        const previousUpdate = this.chainIdIndicesUpdates.at(-1)?.value ?? [1, 10, 137, 288, 42161];
+        // We should now check that previousUpdate is a subset of chainIndices.
+        if (!previousUpdate.every((chainId) => chainIndices.includes(chainId))) {
```

Comment:
WDYT about a stricter check where we check `JSON.stringify(chainIndices.slice(0, previousUpdate.length)) === JSON.stringify(previousUpdate)`?

Reply:
This would require sorting both the `chainIndices` & `previousUpdate` & converting them both to strings. I believe this would result in the same comparison but with a larger time complexity

Reply:
resolved this offline